x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/CacheService.java

```diff
@@ -141,4 +195,125 @@ public void removeFromCache(final Predicate<CacheKey> predicate) {
         }
         cache.refresh();
     }
+
+    void setCacheSyncInterval(TimeValue interval) {
+        cacheSyncTask.setInterval(interval);
+    }
+
+    private void setMaxCacheFilesToSyncAtOnce(int maxCacheFilesToSyncAtOnce) {
+        this.maxCacheFilesToSyncAtOnce = maxCacheFilesToSyncAtOnce;
+    }
+
+    /**
+     * This method is invoked when a {@link CacheFile} notifies the current {@link CacheService} that it needs to be fsync on disk.
+     * <p>
+     * It adds the {@link CacheFile} instance to current set of cache files to synchronize.
+     *
+     * @param cacheFile the instance that needs to be fsync
+     */
+    void onCacheFileUpdate(CacheFile cacheFile) {
+        cacheFilesToSync.offer(cacheFile);
+    }
+
+    /**
+     * This method is invoked after a {@link CacheFile} is evicted from the cache.
+     * <p>
+     * It notifies the {@link CacheFile}'s eviction listeners that the instance is evicted. It also removes the instance from the current
+     * queue of cache files to synchronize if the instance is referenced there.
+     *
+     * @param cacheFile the evicted instance
+     */
+    void onCacheFileRemoval(CacheFile cacheFile) {
+        IOUtils.closeWhileHandlingException(cacheFile::startEviction);
+        cacheFilesToSync.remove(cacheFile);
```

Comment:
Also, calling `remove` (`O(n)`) on a `ConcurrentLinkedQueue` feels like it may bring trouble if we're dealing with a large number of files queued up? Do we even need to do this when we could simply skip evicted files when polling the fsync queue?

Reply:
Agree - Henning also raised this point. The `CacheFile#fsync()` method should return an empty set of ranges if the cache file is evicted and deleted from disk, we can rely on this to skip removed files.

Reply:
I removed the `remove()` call in eabe56292858d6a23a4b78d77eba1b6b6bf998f9